openassessment/xblock/static/js/src/lms/components/WaitingStepDetailsPage.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Alert, Container, Row, Spinner,
+} from '@edx/paragon';
+import WaitingStepList from './WaitingStepList';
+
+const WaitingStepPage = ({ waitingStepDataUrl }) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [waitingStepDetails, setWaitingStepDetails] = useState({
+    display_name: '', must_be_graded_by: '', must_grade: '', student_data: [],
+  });
+
+  const fetchWaitingStepDetails = async () => {
+    // Clear error and display loading component
+    setLoading(true);
+    setError(false);
+    try {
+      // Retrieve waiting step data and decode JSON
+      const response = await fetch(waitingStepDataUrl);
+      const json = await response.json();
+      // Set table data
+      setWaitingStepDetails(json);
+    } catch (requestError) {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchWaitingStepDetails();
+  }, []);
+
+  const waitingStepPageContent = () => {
```

Comment:
This feels like it should be its own component

Reply:
@muselesscreator Agreed, I've moved this to a separate component.